src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,202 @@
+    InterestRateStrategyLike public immutable interestStrategy;
+    TokenLike                public immutable stableDebt;
+    TokenLike                public immutable variableDebt;
+    address                  public immutable adai;
+    address                  public immutable dai;
+
+    uint256 public bar;  // Target Interest Rate [ray]
+
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MAaveDaiPlan/not-authorized");
+        _;
+    }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+
+    constructor(address dai_, address pool_) public {
+        dai = dai_;
+        pool = LendingPoolLike(pool_);
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPlan/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPlan/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPlan/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDaiPlan/invalid-interestStrategy");
+
+        adai = adai_;
+        stableDebt = TokenLike(stableDebt_);
+        variableDebt = TokenLike(variableDebt_);
+        interestStrategy = InterestRateStrategyLike(interestStrategy_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    uint256 constant RAY  = 10 ** 27;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "D3MAaveDaiPlan/overflow");
+    }
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "D3MAaveDaiPlan/underflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPlan/overflow");
+    }
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bar") {
+            require(data <= maxBar(), "D3MAaveDaiPlan/above-max-interest");
+
+            bar = data;
+        } else revert("D3MAaveDaiPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function maxBar() public view returns (uint256) {
```

Comment:
Don't think this should be a function, it's used only once so can just inline it (for example we don't have an equivalent in compound).

Reply:
I was going to do that, and discovered that we should be using it someplace else. Are you okay with this: https://github.com/makerdao/dss-direct-deposit/pull/31/commits/974a0de8213635f399d9af79770a3e6bcbee41ab

Reply:
Yeah, that's fine, an internal function is better.